x/shield/keeper/purchase.go

```diff
@@ -162,7 +162,8 @@ func (k Keeper) RemoveExpiredPurchases(ctx sdk.Context) {
 					purchaseList.Entries = append(purchaseList.Entries[:i], purchaseList.Entries[i+1:]...)
 					pool, err := k.GetPool(ctx, purchaseList.PoolID)
 					if err != nil {
-						panic(err)
+						// skip purchases of closed pools
```

Comment:
Can we get this processed before pool closure to avoid such ignoring? Using error in normal execution path is a bad idea.

Reply:
We need to iterate all the purchase queue if we want to do it before pool closure. We can do it if you think it's acceptable. Or maybe we can return `found` rather than `error`.

Reply:
Updated the return of `GetPool()` from `err` to `found`.